ipatests/pytest_ipa/integration/create_keycloak.py

```diff
@@ -157,3 +157,20 @@ def setup_keycloak_client(host):
                       "-s", "secret={0}".format(password)]
                      )
     time.sleep(60)
+
+
+def uninstall_keycloak(host):
+    key = os.path.join(paths.OPENSSL_PRIVATE_DIR, "keycloak.key")
+    crt = os.path.join(paths.OPENSSL_PRIVATE_DIR, "keycloak.crt")
+    keystore = os.path.join(paths.OPENSSL_PRIVATE_DIR, "keycloak.store")
+
+    host.run_command(["systemctl", "stop", "keycloak"], raiseonerr=False)
+    host.run_command(["getcert", "stop-tracking", "-k", key, "-f", crt],
+                     raiseonerr=False)
+    host.run_command(["rm", "-rf", "/opt/keycloak",
+                      "/etc/sysconfig/keycloak",
+                      "/etc/systemd/system/keycloak.service",
+                      key, crt, keystore])
+    host.run_command(["systemctl", "daemon-reload"])
+    host.run_command(["userdel", "keycloak"])
+    host.run_command(["groupdel", "keycloak"], raiseonerr=False)
```

Comment:
tasks.restore_files(host) should cover that right?  I added backup_file in the setup functions earlier in the file.

Reply:
Yes, `tasks.restore_files()` can be used. But this method does not support incremental backup/restore of the same file, and the code is currently using it this way:
1/ setup_keycloakserver: backup initial /etc/bashrc into /ipatests/file_backup/etc/bashrc
2/ setup_keycloak_scim_plugin: backup /etc/bashrc into /ipatests/file_backup/etc/bashrc = overwrites the backup from step 1 with a new one. The initial version is lost.
3/ uninstall_keycloak: calls restore = restores the version from step 2
4/ uninstall_scim_plugin: calls restore, but restore has already been executed, no files left to restore

To avoid this, the code can do a single backup/restore, in the methods setup_keycloakserver and uninstall_keycloak.

Reply:
Ok, I removed the backup/restore from the plugin setup/uninstall and left it only in the keycloak setup/uninstall.